charon/storage.py

```diff
@@ -698,32 +698,37 @@ def list_folder_content(self, bucket_name: str, folder: str) -> List[str]:
            not in its subfolders.
         """
         bucket = self.__get_bucket(bucket_name)
+
         try:
             if not folder or folder.strip() == "/" or folder.strip() == "":
-                result = bucket.meta.client.list_objects(
+                paginator = bucket.meta.client.get_paginator('list_objects_v2')
+                pages = paginator.paginate(
                     Bucket=bucket.name,
                     Delimiter='/'
                 )
             else:
                 prefix = folder if folder.endswith("/") else folder+"/"
-                result = bucket.meta.client.list_objects(
+                paginator = bucket.meta.client.get_paginator('list_objects_v2')
```

Comment:
I think this paginator only needs one line declaration just below "try:", which can be reused.

Reply:
fixed